### PR TITLE
Upgrade yarn to 1.21.1

### PR DESCRIPTION
**Pull request by vzhukovskii**
### What does this PR do?
This changes proposal upgrades yarn from version **1.17.3** to **1.21.1** inside `che-theia-dev` and `che-theia` images. Also due to breaking changes in yarn **1.21.1** related to support `:` char in bin entry names, there is a proposal to update bin entry provided by theia-generator from `che:theia` to `che-theia`. Also there is an updates in docs and related devfiles.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/15771

#### Release Notes
Upgrade yarn from **1.17.3** to **1.21.1** in `che-theia-dev` and `che-theia` images.

#### Docs PR
N/A

### CODE REVIEW

File: generator/README.md [hunk 1]
```diff
@@ -49,22 +49,22 @@ In order to build the product, just run `yarn` at the root folder (where theia h
 
 ### Custom extension and plugin set
 
-Also you can provide custom `yaml` with your extension set, by using `-c` or `--config` parameter of `che:theia init` :
+Also you can provide custom `yaml` with your extension set, by using `-c` or `--config` parameter of `che-theia init` :
 
-`che:theia init -c ./path/to/custom/che-theia-init-sources.yaml`
+`che:-heia init -c ./path/to/custom/che-theia-init-sources.yaml`
```
Comment (vzhukovskii) [inline generator/README.md]: Fix spelling

File: dockerfiles/theia-dev/docker/alpine/setup-yarn.dockerfile [hunk 2]
```diff
@@ -1 +1,10 @@
-# nothing to do on alpine
+ENV YARN_VERSION=1.21.1
```
Comment (benoitf) [inline dockerfiles/theia-dev/docker/alpine/setup-yarn.dockerfile]: maybe we just need to bump to https://github.com/nodejs/docker-node/blob/eb89a38195d4e289812204847baf06c6c53406c0/10/alpine3.9/Dockerfile

10.19.0-alpine3.11

Comment (vzhukovskii) [reply dockerfiles/theia-dev/docker/alpine/setup-yarn.dockerfile]: It works. Thanks!

File: dockerfiles/theia/docker/ubi8/runtime-setup-yarn.dockerfile [hunk 3]
```diff
@@ -1,2 +1,2 @@
 # install yarn dependency
-RUN npm install -g yarn@1.17.3
+RUN npm install -g yarn@1.21.1
```
Comment (azatsarynnyy) [inline dockerfiles/theia/docker/ubi8/runtime-setup-yarn.dockerfile]: fyi @nickboldt 
maybe it affects crw build

File: dockerfiles/theia-dev/docker/ubi8/setup-yarn.dockerfile [hunk 4]
```diff
@@ -1,2 +1,2 @@
 # install yarn dependency
-RUN npm install -g yarn@1.17.3
+RUN npm install -g yarn@1.21.1
```
Comment (azatsarynnyy) [inline dockerfiles/theia-dev/docker/ubi8/setup-yarn.dockerfile]: fyi @nickboldt 
maybe it affects crw build

File: dockerfiles/theia/docker/ubi8/runtime-setup-yarn.dockerfile (hunk 3 above)
Comment (vzhukovskii) [reply dockerfiles/theia/docker/ubi8/runtime-setup-yarn.dockerfile]: I guess, it shouldn't, but need to ensure

File: dockerfiles/theia-dev/docker/ubi8/setup-yarn.dockerfile (hunk 4 above)
Comment (nickboldt) [reply dockerfiles/theia-dev/docker/ubi8/setup-yarn.dockerfile:2]: yes, obviously this affects CRW. Thanks for the headsup (which I'm only seeing now) :(

**Pull request merged**